runtime/plugin/manpager.vim

```diff
@@ -10,17 +10,17 @@ endif
 command! -nargs=0 MANPAGER call s:MANPAGER() | delcommand MANPAGER
 
 function! s:MANPAGER()
-  let page_pattern = '\v\w+%([-_.]\w+)*'
+  let page_pattern = '\v\w[-_.:0-9A-Za-z]*'
   let sec_pattern = '\v\w+%(\+\w+)*'
   let pagesec_pattern = '\v(' . page_pattern . ')\((' . sec_pattern . ')\)'
 
   if $MAN_PN is '1'
-    let manpage = matchstr( getline(nextnonblank(1)), '^' . pagesec_pattern )
+    let manpage = tolower(matchstr( getline(nextnonblank(1)), '^' . pagesec_pattern ))
```

Comment:
That seems like just moving the problem around. Wouldn't it be better to use case-insensitive RE instead?

Reply:
If `$MAN_PN` is '1' then yes, the problem is only moved around.
If however `$MAN_PN` is different from 1, then the problem is solved.
This patch comes with a documentation update that recommends not setting `MAN_PN = 1` if the man implementation does it, like `man-db`, which is used on most Linux systems. See the discussion at

https://groups.google.com/forum/#!topic/vim_dev/pWZmt_7GkxI

Reply:
The necessity for using `tolower()` in the first place comes from the fact that most manpages put their title on the page itself in capital letters, but their file name in lower case letters. It is the latter that is used when invoking `man`.

Reply:
> their file name in lower case letters.

Except for the case when they don't (somebody mentioned ``Xorg(1)`` which on my system is in ``/usr/share/man/man1/Xorg.1.gz``. So, I guess
```viml
let pagesec_pattern = '\v\c(' . page_pattern . ')\((' . sec_pattern . ')\)'
```
would work better, won't it?

Reply:
In this case yes, but then there are plenty other examples, for example

man ls

is mentioned in the thread, that put their names in capital letters, LS(1) but whose file name is in lower case letters.
These man pages, whose file names are all lower case letters but whose name is given in capital letters, are far more common than man pages whose file name contains capital letters and that correctly give their name in the man page itself.

Knowing which particular man pages correctly give their file name in the man page and which not is a losing proposition, as said in the thread, and this pull request agrees on it.

In this case, all you can hope for is that you use a common Linux system or write your proper wrapper script outside of Vim.

Reply:
Regarding the regular expression, it is already agnostic to lower or upper cases, so the addition of `\c` changes nothing.

Reply:
Oh, sorry.

Reply:
> Regarding the regular expression, it is already agnostic to lower or upper cases, so the addition of \c changes nothing.

@Konfekt still, I don't get it. If RE is case agnostic, why do we need to do ``lower()`` at all then?

Reply:
Because common are man pages such as

  man ls

that put their names in the title in capital letters, LS(1), but whose file name is in lower case letters.

In fact, these man pages, whose file names are all lower case letters but whose name is given in capital letters, are far more common than man pages whose file name contains capital letters and that correctly give their name in the man page itself.
Therefore, `tolower()` is far more often a cure than a plague, and worth a shot.